sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1203,6 +1207,32 @@ def check_datapath_state(self, api, channel, states):
 
         return done
 
+    def get_configured_laser_freq(self, lport):
+        """
+           Return the Tx power configured by user in CONFIG_DB's PORT table
+        """
+        freq = 0
+        asic_index = self.port_mapping.get_asic_id_for_logical_port(lport)
+        port_tbl = self.xcvr_table_helper.get_cfg_port_tbl(asic_index)
+
+        found, port_info = port_tbl.get(lport)
+        if found and 'laser_freq' in dict(port_info):
+            freq = dict(port_info)['laser_freq']
```

Comment:
what if found is True but no frequency is configured in port_info? It's likely to happen when a ZR module is plugged but freq config is missing. Do we see the port to be err-disabled?

Reply:
the port_info{} dictionary represents the PORT table in CONFIG_DB. If laser frequency is NOT configured, then we move on with the module's default frequency.

Reply:
freq init value == 0 in this function. Does it mean when we see freq == 0, it will set module frequency to the default frequency?

Reply:
yes module comes with default frequency if not configured